xul/bootstrap.js

```diff
@@ -0,0 +1,308 @@
+
+const branch = "extensions.esrc-explorer.";
+const apmo_pattern = /^https?:\/\/addons\.palemoon\.org\/addon\/(.*?)\//;
+const apmo_download_pattern = /^https?:\/\/addons\.palemoon\.org\/\?component=download&id=(.+?)&version=(.+?)&hash/;
+const apmo_match_pattern = '*://addons.palemoon.org/?component=download*';
+
+let styleSheetService = Cc["@mozilla.org/content/style-sheet-service;1"].getService(Ci.nsIStyleSheetService);
+let styleSheetURI = Services.io.newURI("chrome://esrc-explorer/skin/style.css", null, null);
+
+let gWindowListener, navigator;
+
+function ESrcExplorer(aWindow) {
+  this.init(aWindow);
+}
+ESrcExplorer.prototype = {
+
+  init: function(aWindow) {
+    this.browserWindow = aWindow;
+    this.tabBrowser = aWindow.gBrowser;
+
+    if (Services.prefs.getBoolPref(branch + "showButton")) {
+      this.addButton();
+    }
+
+    if (Services.prefs.getBoolPref(branch + "showContext")) {
+      this.addContext();
+    }
+
+    this.prefBranch = Services.prefs.getBranch(branch);
+    this.prefBranch.addObserver("", this, false);
+  },
+
+  done: function() {
+    this.prefBranch.removeObserver("", this);
+    this.prefBranch = null;
+
+    if (Services.prefs.getBoolPref(branch + "showButton")) {
+      this.removeButton();
+    }
+
+    if (Services.prefs.getBoolPref(branch + "showContext")) {
+      this.removeContext();
+    }
+
+    this.tabBrowser = null;
+    this.browserWindow = null;
+  },
+
+  observe: function(aSubject, aTopic, aData) {
+    if (aTopic != "nsPref:changed") return;
+    switch (aData) {
+      case 'showButton':
+        if (Services.prefs.getBoolPref(branch + "showButton")) {
+          this.addButton();
+        } else {
+          this.removeButton();
+        }
+        break;
+      case 'showContext':
+        if (Services.prefs.getBoolPref(branch + "showContext")) {
+          this.addContext();
+        } else {
+          this.removeContext();
+        }
+        break;
+    }
+  },
+
+  openExplorerOrSave: function(aUrl, aSave) {
+    let crx_url, filename;
+    if (apmo_pattern.test(aUrl)) {
+      crx_url = this.tabBrowser.getBrowserForTab(this.tabBrowser.selectedTab).
+                  contentDocument.getElementsByClassName("dllink_green")[0].href;
+    } else if (apmo_download_pattern.test(aUrl)) {
+      crx_url = aUrl;
+    } else {
+      crx_url = get_crx_url(aUrl);
+    }
+    if (!crx_url) {
+      Cu.reportError('Cannot find extension URL');
+      return;
+    }
+    let match1 = apmo_download_pattern.exec(crx_url);
+    if (match1) {
+      let match2 = apmo_pattern.exec(this.tabBrowser.selectedTab.linkedBrowser.currentURI.spec);
+      if (match2) {
+        filename = match2[1] + "-" + match1[2] + ".zip";
+      } else {
+        filename = match1[1] + "-" + match1[2] + ".zip";
+      }
+    } else {
+      filename = get_zip_name(crx_url);
+    }
+    if (aSave) {
+      this.browserWindow.internalSave(crx_url, null, filename, null, "application/zip",
```

Comment:
The "Download" feature of CRXViewer is a bit more sophisticated than this. Chrome extensions are almost zip files. They have some header prepended to the actual zip data. This is removed by CRXViewer.

You're just downloading the file as-is. Either change the logic to match with the other CRXViewer extensoons, or add a comment to emphasize the difference in behavior.